### PR TITLE
Fix modpow for exp=1

### PR DESCRIPTION
**Pull request by youknowone**
Unfortunately, there was cases modulus never happened

### CODE REVIEW

File: src/biguint.rs [hunk 1]
```diff
@@ -2285,6 +2285,9 @@ fn plain_modpow(base: &BigUint, exp_data: &[BigDigit], modulus: &BigUint) -> Big
 
     let mut exp_iter = exp_data[i + 1..].iter();
     if exp_iter.len() == 0 && r.is_one() {
+        if &base > modulus {
+            base %= modulus;
+        }
```
Comment (cuviper) [inline src/biguint.rs]: I think it would be better to normalize this above where we create a `base.clone()`, just use `base % modulus` instead. The implementation of that `%` operator already checks if the value is less than the modulus too.

File: tests/modpow.rs [hunk 2]
```diff
@@ -81,6 +81,7 @@ mod biguint {
         check_modpow::<u32>(0, 15, 11, 0);
         check_modpow::<u32>(3, 7, 11, 9);
         check_modpow::<u32>(5, 117, 19, 1);
+        check_modpow::<u32>(20, 1, 2, 0);
```
Comment (cuviper) [inline tests/modpow.rs]: Please add a similar test with an odd modulus too, so that corner case is not missed. (But I think that implementation is already OK.)

Comment (youknowone) [reply tests/modpow.rs]: Additionally, I added api surface test for small values below

**Comment by cuviper** [comment]
Thanks!

bors r+
